Agents/PropertyValueEstimationAgent/tests/conftest.py

```diff
@@ -0,0 +1,380 @@
+        # Retrieve host and port from RDB URL assuming default format like
+        # jdbc:postgresql://localhost:5432/<url_route>
+        host = rdb_url.split(':')[2].replace('//', '')
+        port = rdb_url.split(':')[3].split('/')[0]
+        return pg.connect(host=host, port=port, database=DATABASE,
+                          user=DB_USER, password=DB_PASSWORD)
+
+
+def initialise_timeseries(kgclient, dataIRI, dates, values, rdb_url, 
+                          rdb_user, rdb_password):
+    # Initialise timeseries (in Blazegraph and PostgreSQL) with given dates and values
+    # Initialise time series client
+    ts_client = TSClient(kg_client=kgclient, rdb_url=rdb_url, rdb_user=rdb_user, 
+                         rdb_password=rdb_password)
+    # Create time series from test data                        
+    ts = TSClient.create_timeseries(dates, [dataIRI], [values])
+    
+    with ts_client.connect() as conn:
+        # Initialise time series in Blazegraph and PostgreSQL
+        ts_client.tsclient.initTimeSeries([dataIRI], [DATACLASS], TIME_FORMAT_SHORT, conn)
+        # Add test time series data
+        ts_client.tsclient.addTimeSeriesData(ts, conn)
+
+
+def retrieve_timeseries(kgclient, dataIRI, rdb_url, rdb_user, rdb_password):
+    # Initialise time series client
+    ts_client = TSClient(kg_client=kgclient, rdb_url=rdb_url, rdb_user=rdb_user, 
+                         rdb_password=rdb_password)
+    with ts_client.connect() as conn:
+        ts = ts_client.tsclient.getTimeSeries([dataIRI], conn)
+    dates = ts.getTimes()
+    # Unwrap Java time objects
+    dates = [d.toString() for d in dates]
+    values = ts.getValues(dataIRI)
+    return dates, values
+
+
+def get_marketvalue_details(sparql_client, market_value_iri):
+    # Returns details associated with Market Value instance (OM:AmountOfMoney)
+    query = f"""
+        SELECT ?value ?input_iri ?input_type
+        WHERE {{
+        <{market_value_iri}> <{RDF_TYPE}> <{OM_AMOUNT_MONEY}> ; 
+                             <{OM_HAS_VALUE}> ?measure ; 
+                             <{ONTODERIVATION_BELONGSTO}>/<{ONTODERIVATION_ISDERIVEDFROM}> ?input_iri . 
+        ?measure <{RDF_TYPE}> <{OM_MEASURE}> ;  
+                 <{OM_NUM_VALUE}> ?value . 
+        ?input_iri <{RDF_TYPE}> ?input_type . 
+        }}
+        """
+    response = sparql_client.performQuery(query)
+    if len(response) == 0:
+        return None
+    else:
+        # Derivation inputs (i.e. isDerivedFrom)
+        key = set([x['input_type'] for x in response])
+        inputs = {k: [x['input_iri'] for x in response if x['input_type'] == k] for k in key}
+        # Market Value
+        market_value = list(set([float(x['value']) for x in response]))
+
+        return inputs, market_value
+
+
+def get_derivation_status(sparql_client, derivation_iri):
+    # Returns status comment of derivation
+    # TODO: Test for ERROR status code; however not available in pyderivationagent.data_model.iris
```

Comment:
This is fixed in https://github.com/cambridge-cares/TheWorldAvatar/pull/455, can you have a try?

Reply:
Works. [558e706](https://github.com/cambridge-cares/TheWorldAvatar/pull/454/commits/558e7067642012361efa8614912b3dea18633451)